File: src/구현/프로그래머스_70129_이진 변환 반복하기.js
function solution(s) {
    let answer = [];
    let convert = 0,
        removed = 0;

    while (s.length != 1) {
        removed += s.length;
        s = s.replaceAll("0", "");
        removed -= s.length;
        s = s.length.toString(2);
        convert++;
    }

    answer.push(convert, removed);

    return answer;
}

console.log(solution("110010101001"));
console.log(solution("01110"));
console.log(solution("1111111"));

File: src/구현/프로그래머스_70129_이진 변환 반복하기.py
def solution(s):
    convert, zero = 0, 0

    while len(s) != 1:
        zero_cnt = s.count("0")
        s = "1" * (len(s) - zero_cnt)
        zero += zero_cnt
        convert += 1
        s = bin(len(s))[2:]

    return [convert, zero]


print(solution("110010101001"))
print(solution("01110"))
print(solution("1111111"))
